tests/amazon_test.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { test, expect } from '@playwright/test';
+import { searchForProduct } from './utils';
+
+test('Search for gift card on Amazon and apply Netflix filter', async ({ page }) => {
+  await page.goto('https://www.amazon.ca');
+
+  await searchForProduct(page, 'gift card');
+
+  await page.waitForSelector('.s-main-slot');
+
+  const resultsSummaryText = await page.innerText('.s-desktop-toolbar');
+
+  const totalResultsMatch = resultsSummaryText.match(/([\d,]+) results/);
+  const totalResults = totalResultsMatch ? parseInt(totalResultsMatch[1].replace(/,/g, ''), 10) : 0;
```

Comment:
nit: This might be more readable if we pick apart the elements of this expression and store them separately.

For example:

```
  const resultsRegex = /([\d,]+) results/;
  const matches = resultsSummaryText.match(resultsRegex);
  const numberWithCommas = matches ? matches[1] : '0';
  const totalResults = parseInt(numberWithCommas.replace(/,/g, ''), 10);
```

You could even turn this into a function called `getTotalResultsNumber()`. These are all optional stylistic suggestions.